ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -634,11 +646,36 @@ object Serializers {
     }
 
     def readTree(): Tree = {
+      val pos = readPosition()
+      readTreeFromTag(input.readByte())(pos)
+    }
+
+    def readOptTree(): Option[Tree] = {
+      if (useHacks068) {
+        readTreeAllowEmpty()
+      } else {
+        import input._
+        if (readBoolean()) {
+          Some(readTree())
+        } else {
+          None
+        }
+      }
+    }
+
+    private def readTreeAllowEmpty(): Option[Tree] = {
+      require(useHacks068)
+      val pos = readPosition()
+      val tag = input.readByte()
+      if (tag == TagEmptyTree) None
+      else Some(readTreeFromTag(tag)(pos))
```

Comment:
It is indeed unfortunate that the format has the position before the kind. I think we could just hard-code the encoding for `NoPosition` (i.e., `-1.toByte`) when serializing `None`. In 1.0, we can flip the position and kind and get rid of this. WDYT?